keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/api/ec2/ServerGroup.kt

```diff
@@ -17,18 +17,30 @@
  */
 package com.netflix.spinnaker.keel.api.ec2
 
-import com.netflix.spinnaker.keel.api.Monikered
 import com.netflix.spinnaker.keel.model.Moniker
+import com.netflix.spinnaker.keel.model.parseMoniker
+import de.danielbechler.diff.inclusion.Inclusion.EXCLUDED
+import de.danielbechler.diff.introspection.ObjectDiffProperty
 
 data class ServerGroup(
-  override val moniker: Moniker,
+  /**
+   * This field is immutable, so we would never be reacting to a diff on it. If the name differs,
+   * it's a different resource. Also, a server group name retrieved from CloudDriver will include
+   * the sequence number. However, when we resolve desired state from a [ClusterSpec] this field
+   * will _not_ include the sequence number. Having it on the model returned from CloudDriver is
+   * useful for some things (e.g. specifying ancestor server group when red-blacking a new version)
+   * but is meaningless for a diff.
+   */
+  @get:ObjectDiffProperty(inclusion = EXCLUDED)
+  val name: String,
   val location: Location,
   val launchConfiguration: LaunchConfiguration,
   val capacity: Capacity = Capacity(1, 1, 1),
   val dependencies: Dependencies = Dependencies(),
   val health: Health = Health(),
   val scaling: Scaling = Scaling(),
   val tags: Map<String, String> = emptyMap()
-) : Monikered {
-  override val id = "${location.accountName}:${location.region}:${moniker.name}"
```

Comment:
`ServerGroup` should definitely not have an `id`, this is just going to confuse someone in the future.